src/main/java/com/cburch/logisim/std/ttl/Ttl74153.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file is part of logisim-evolution.
+ *
+ * Logisim-evolution is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * Logisim-evolution is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with logisim-evolution. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Original code by Marcin Orlowski (http://MarcinOrlowski.com), 2021.
+ */
+
+package com.cburch.logisim.std.ttl;
+
+import com.cburch.logisim.data.Value;
+import com.cburch.logisim.instance.InstancePainter;
+import com.cburch.logisim.instance.InstanceState;
+import java.util.ArrayList;
+
+/**
+ * TTL 74x153: Dual 4-line to 1-line data selector
+ * Model based on <a href="https://www.ti.com/lit/ds/symlink/sn74ls153.pdf">74LS153 datasheet</a>.
+ */
+public class Ttl74153 extends AbstractTtlGate {
+  /**
+   * Unique identifier of the tool, used as reference in project files.
+   * Do NOT change as it will prevent project files from loading.
+   * Identifier value MUST be unique string among all tools.
+   */
+  public static final String _ID = "74153";
+
+  public static final int DELAY = 1;
+
+  // IC pin indices as specified in the datasheet
+
+  // Inputs
+  public static final byte A = 14;
+  public static final byte B = 2;
+
+  public static final byte L1_G = 1;
+  public static final byte L1_C0 = 6;
+  public static final byte L1_C1 = 5;
+  public static final byte L1_C2 = 4;
+  public static final byte L1_C3 = 3;
+
+  public static final byte L2_G = 15;
+  public static final byte L2_C0 = 10;
+  public static final byte L2_C1 = 11;
+  public static final byte L2_C2 = 12;
+  public static final byte L2_C3 = 13;
+
+  // Outputs
+  public static final byte L1_Y = 7;
+  public static final byte L2_Y = 9;
+
+  // Power supply
+  public static final byte GND = 8;
+  public static final byte VCC = 16;
+
+  private InstanceState _state;
+
+  public Ttl74153() {
+    super(
+            _ID,
+            (byte) 16,
+            new byte[] {
+              L1_Y, L2_Y
+            },
+            new String[] {
+              "n1G", "B", "1C3", "1C2", "1C1", "1C0", "1Y",
+              "2Y", "2C0", "2C1", "2C2", "2C3", "A", "n2G"
+            },
+            null);
+  }
+
+  @Override
+  public void paintInternal(InstancePainter painter, int x, int y, int height, boolean up) {
+    // As tooltips can be longer than what can fit as pin name while painting IC internals,
+    // we need to shorten it first to up to 4 characters to keep the diagram readable.
+    final var label_len_max = 4;

```

Comment:
Please use camelCase.

Reply:
I did one better than that: I've added a meaningful internal structure diagram